### ISSUE TITLE
Feature request: ability to slow down tests for debugging

### ISSUE THREAD

**Issue by gillyb** [theintern/intern#613]
When a test fails, and I would like to debug it, it would help if i can add some configuration that will add an automatic delay between each command so I can view the UI move a little slower.

**Comment by sholladay** [comment]
Debugging could definitely be better. But you'd probably still need to add `.sleep()` calls within functional tests, unless you're talking about slowing every WebDriver command by multiple seconds. And at that point it would be unbearably slow. So I'm not sure adding general configuration to do this would really help so much as it would just make you want to go get a coffee while its running.

**Comment by gillyb** [comment]
You're right, although I still feel like running my tests locally happen so fast that it's hard to tell what's happening. I find myself adding various `sleep()` calls along with `console.log()` with information between different actions, until I manage to find out where exactly my test is breaking.

If we had a setting like this, then maybe it will be possible to add a helper method that will change the timeout configuration for each WebDriver command in a certain scope, so I'll be able to do something like this :

```javascript
bdd.it('my test', function() {
    let remoteDriver = this.remote;
    return remoteDriver
        .findByCssSelector('...')
        .click().type('something').end()
        // here starts the part i want to slow down by 1000ms for each WebDriver command
        .slowdown(200, function() {
            return remoteDriver
                .findByCssSelector('...')
                .click()
                .acceptAlert()
                .end();
        })
        // more stuff that can happen fast...
        .findByCssSelector('...').click().end()
});
```

Don't know... just a thought that might really help. :)

**Comment by jason0x43** [comment]
You can also use node-inspector to [step through functional tests](https://www.sitepen.com/blog/2014/05/23/how-can-i-debug-intern-tests/)

**Comment by sholladay** [comment]
Mmmm having a helper method that allows you to add one little chunk rather than sprinkling `sleep()` all over the place sounds like an excellent idea to me. Let's call it `molasses()` :). It would be great if it didn't even need to wrap any other calls. I wonder how much effort that would be.
